src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -266,6 +268,63 @@ public async Task<Document> ApplyCodeFixesForSpecificDiagnosticIdAsync(Document
             return solution.GetDocument(document.Id) ?? throw new NotSupportedException(EditorFeaturesResources.Removal_of_document_not_supported);
         }
 
+        private bool TryGetWorkspaceFixersMap(Document document, out Lazy<ImmutableDictionary<DiagnosticId, ImmutableArray<CodeFixProvider>>> fixerMap)
+        {
+            if (_lazyWorkspaceFixersMap == null)
+            {
+                var workspaceFixersMap = _getWorkspaceFixersMap(document.Project.Solution.Workspace);
+                Interlocked.CompareExchange(ref _lazyWorkspaceFixersMap, workspaceFixersMap, null);
+            }
+
+            return _lazyWorkspaceFixersMap.TryGetValue(document.Project.Language, out fixerMap);
+        }
+
+        private bool TryGetWorkspaceFixersPriorityMap(Document document, out Lazy<ImmutableDictionary<CodeFixProvider, int>> fixersPriorityMap)
+        {
+            if (_lazyFixerPriorityMap == null)
+            {
+                var fixersPriorityByLanguageMap = _getFixerPriorityMap(document.Project.Solution.Workspace);
+                Interlocked.CompareExchange(ref _lazyFixerPriorityMap, fixersPriorityByLanguageMap, null);
+            }
+
+            return _lazyFixerPriorityMap.TryGetValue(document.Project.Language, out fixersPriorityMap);
+        }
+
+        private static bool TryGetWorkspaceFixer(
+            Lazy<CodeFixProvider, CodeChangeProviderMetadata> lazyFixer,
+            Workspace workspace,
+            bool showInfoBarOnException,
+            [NotNullWhen(returnValue: true)] out CodeFixProvider? fixer)
+        {
+            try
+            {
+                fixer = lazyFixer.Value;
+                return true;
+            }
+            catch (Exception ex)
+            {
+                // Gracefully handle exceptions in creating fixer instance.
+                // Show info bar for exception, if needed.
+                if (showInfoBarOnException)
+                {
+                    var errorReportingService = workspace.Services.GetRequiredService<IErrorReportingService>();
```

Comment:
doesn't look like the `VisualStudioErrorReportingService` which this is going to call in VS' case will log failure telemetry. Is that something we want to do for this case?

Reply:
Seems like that should be an independent change made to `VisualStudioErrorReportingService`, as we probably want telemetry from all clients invoking this?

Reply:
Actually, it seems the CodeFixService already MEF imports the error loggers, so I will update the code to log exception here. Thanks!